### ISSUE TITLE
Flag for reproducibity

### ISSUE THREAD

**Issue by yipliu** [ashleve/lightning-hydra-template#341]
Hi,

I have used the fixed seed, but the optimal hyperparameter I obtained by hydra search could not reproduce the experimental results. It's important for open source.

I notice there was the same issue in [here](https://github.com/PyTorchLightning/pytorch-lightning/discussions/11250). It looks like the following parameters need to be set in **Trainer**. I am also trying this method now.
```python
# default used by the Trainer
trainer = Trainer(deterministic=False)
```

If this phenomenon is common, I suggest that it be explicitly stated in the **readme** file.

**Comment by ashleve** [comment]
Thanks, good suggestion. For now I added it to default trainer flags here
https://github.com/ashleve/lightning-hydra-template/pull/342/files

**Comment by yipliu** [comment]
Hi ashleve,

Can we discuss the empirical of tuning hyperparameters?

For a model that is more complex than the template model, I set the seed and set deterministic=True. However, I was not able to reproduce the performance of my model.

The hyperparameters are obtained by  **_hydra  search_** and the performance is great on **_dev sets_** but was poor on the **_test set_**.  (Some general tips I have also set, such as earlystop)

Could you share your opinion on this?
